### ISSUE TITLE
`eegplot()` extremly slow when plotting later epochs

### ISSUE THREAD

**Issue by mattansb** [eeglabdevelopers/eeglab#108]
### Description

Since switching to V2019.1, the data scrolling via `eegplot` has become extremely slow - whereas in previous versions plotting was more-or-less instantaneous when plotting the next couple of trials, now takes many seconds to plot (more than 20 seconds to plot 20 trials) - which is honestly unbearable.

#### Steps to Reproduce

We've tried on 3 computers, with different sized data sets - in all the same slowness in experienced.

#### Expected behavior:

Ability to scroll epoched data fast.

#### Actual behavior:

Data scrolling is painstakingly slow.

#### Versions

|   |   |
|----| ---- | 
| OS version     | Win10 |
| Matlab version | R2017b |
| EEGLAB version | 2019.1 |

**Comment by nucleuscub** [comment]
Hi @mattansb,
I'll try with Win10, but so far with MACOS there is no issue at all. There has not been any change in eegplot.m between the 2019.0 and 2019.1 to justify the slowdown.
Question here:
Are the three computers you tried running on windows?
What was the previous EEGLAB version installed and running without issues?
Thks,
R

**Comment by nucleuscub** [comment]
One more... how many events do you have in the dataset?

**Comment by arnodelorme** [comment]
Yes, the last change to eegplot.m is dated of April 2019 which was before 2019.0 was released. If you can run the profiler, we can assess what is slowing things down. My guess is a problem between Matlab and Windows display drivers. Maybe change the default renderer for figures.

set(0, 'defaultFigureRenderer', 'painters')

or

set(0, 'defaultFigureRenderer', 'zbuffer')

Arno

**Comment by mattansb** [comment]
@nucleuscub and @arnodelorme 

It seems the problem is different than I thought: **The *later* the epoch, the *longer* it takes to plot it.** That is, clicking `>>` from 1 to 6 plots faster (less than 2 seconds) than from 701 to 706 (~20 seconds).
[I've changed the name of the issue accordingly]

- I've tried this on eeglab 14.1, and the problem was there too. We just didn't notice it because previous projects had a smaller number of epochs.
- All three computers we tried are running on windows.
- Trying `set(0, 'defaultFigureRenderer', 'painters')` / `set(0, 'defaultFigureRenderer', 'zbuffer')` had no effect.

**Comment by nucleuscub** [comment]
Thanks, @mattansb , 
 Can you please make a profile of the scrolling on eegplot.
 From the matlab command line activate the profile (profile on), then launch eegplot and start scrolling as to reproduce the issue. When done,  use 'profile review' to generate the report. Please share that with us. I'm afraid the function axes may be taking too long for some of the graphical processing in Windows. In any case, let's see the profile first.
 Thanks,
Ramon

**Comment by mattansb** [comment]
@nucleuscub thanks!
Here is a zip of the following:

```Matlab
profile on
eeglab
% open data scroll and scroll a couple of times
% jump to trial 300 and scroll a couple of times
% jump to trial 700 and scroll a couple of times
profile viewer
profsave
```

[profile_results.zip](https://github.com/sccn/eeglab/files/4039445/profile_results.zip)

**Comment by arnodelorme** [comment]
This is because you have a lot of events. Most of the time the function is evaluating involve plotting events. I am not sure about the proportional time with how far it is in the data. This does nor make sense to me.

Would you mind to try the EEGbrowser plugin and tell us if has the same problem?

**Comment by mattansb** [comment]
`EEGbrowser` browser does not have the same problem, but it is not suitable for my needs as I need to be able to see the trial number (#107 ) which `EEGbrowser` does not show (only seconds). (Also the mouse-hover features from `pop_eegbrowser` would be useful here as well.)

**Comment by arnodelorme** [comment]
Would you mind to share a dataset for testing purposes.

Arno

**Comment by mattansb** [comment]
Sure:
- [601_fil_seg.set](https://drive.google.com/a/post.bgu.ac.il/file/d/1kB6ORp3pe2k6gwHbZOV1iIHKpvhuxo6k/view?usp=sharing)
- [601_fil_seg.fdt](https://drive.google.com/a/post.bgu.ac.il/file/d/1kAiESgpictVDB83XCBAy8rHH3Wx2wxhW/view?usp=sharing)

**Comment by arnodelorme** [comment]
I have corrected a bug in c600a5376 associated with scanning all events. You can do a git pull or simply replace the file https://github.com/sccn/eeglab/blob/develop/functions/sigprocfunc/eegplot.m

Arno

**Comment by mattansb** [comment]
@arnodelorme Thank you so much! Works great!
